File: Basic Programming/Hackerrank 30 days of code python3 solution/Day_28.py
#                      Name : Arif Khan
#                      Judge: HACKERRANK
#                      University: Primeasia University
#                      problem: Day 28: RegEx, Patterns, and Intro to Databases
#                      Difficulty: Easy
#                      Problem Link: https://www.hackerrank.com/challenges/30-regex-patterns/problem
#

#!/bin/python3

import math
import os
import random
import re
import sys

if __name__ == '__main__':
    N = int(input())
    firstName = []

    for N_itr in range(N):
        firstNameEmailID = input().split()
        emailId = firstNameEmailID[1]
        if emailId.find("@gmail.com") != -1:
            firstName.append(firstNameEmailID[0])

    for i in sorted(firstName):
        print(i)

File: Basic Programming/Hackerrank 30 days of code c++ solution/Day_28.cpp
/*                    Name : Arif Khan
                      Judge: HACKERRANK
                      University: Primeasia University
                      problem: Day 28: RegEx, Patterns, and Intro to Databases
                      Difficulty: Easy
                      Problem Link: https://www.hackerrank.com/challenges/30-regex-patterns/problem
*/

#include <bits/stdc++.h>
using namespace std;

int main(){
    int N;
    cin >> N;
    multiset<pair<string,string>>ms;
    multiset<pair<string,string>>::iterator it;
    for(int a0 = 0; a0 < N; a0++){
        string firstName;
        string emailID;
        string s;
        size_t p;
        cin >> firstName >> emailID;
        p = emailID.find('@');
        s = emailID.substr(p+1);
        ms.insert(make_pair(firstName,s));
    }
    for(it = ms.begin();it != ms.end(); it++){
        if(it->second == "gmail.com")
            cout<<it->first<<endl;
    }
    return 0;
}